src/Symfony/Component/JsonEncoder/Attribute/ValueTransformer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\Attribute;
+
+use Symfony\Component\JsonEncoder\Exception\LogicException;
+
+/**
+ * Defines a callable or a {@see \Symfony\Component\JsonEncoder\ValueTransformer\ValueTransformerInterface} service id
+ * that will be used to transform the property data during encoding and decoding.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY)]
+class ValueTransformer
+{
+    private ?\Closure $toPhpValue;
+    private ?\Closure $toJsonValue;
+
+    /**
+     * @param (callable(mixed, array<string, mixed>?): mixed)|(callable(mixed): mixed)|null $toPhpValue
+     * @param (callable(mixed, array<string, mixed>?): mixed)|(callable(mixed): mixed)|null $toJsonValue
+     */
+    public function __construct(
+        private ?string $serviceId = null,
+        mixed $toPhpValue = null,
+        mixed $toJsonValue = null,
```

Comment:
Why `mixed`? The PHPDoc says the values must be callable.

```suggestion
        ?callable $toPhpValue = null,
        ?callable $toJsonValue = null,
```

Reply:
Indeed, there is no need to keep mixed here, updated.